### ISSUE TITLE
[expo-updates] Self hosted updates doesn't work

### ISSUE THREAD

**Issue by RodolfoGS** [expo/expo#13872]
### Summary

I'm upgrading my app to SDK 42 and `expo-updates: 0.8.1` and I have issues with expo-updates.
I'm using the self hosted updates workflow and the `Updates.fetchUpdateAsync` method always return that haven't an update available (`{isNew: false}`).

I tested the connections from my device, and I notice that my custom server url (`__public_server__/ios-index.json`) never is called.

If you use the expo server (using the `expo publish` command) works great. But not when you use your own servers (`expo export...` command)

I created a reproducible example. This happen on iOS and Android.

My demo App.js:
```
import { StatusBar } from 'expo-status-bar';
import React from 'react';
import { Button, StyleSheet, Text, View } from 'react-native';
import * as Updates from 'expo-updates';

export default function App() {
  handleCheckUpdate = async () => {
    try {
      const response = await Updates.fetchUpdateAsync();
      alert(JSON.stringify(response)); // always return {isNew: false}
    } catch (error) {
      alert(JSON.stringify(error.message));
    }
  }

  handleForceUpdate = async () => {
    await Updates.reloadAsync();
  }

  return (
    <View style={styles.container}>
      <Text>Release: {Updates.manifest.revisionId}</Text>
      <Button onPress={handleCheckUpdate} title="Check Update" />
      <Text />
      <Button onPress={handleForceUpdate} title="Force Update" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
```

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

bare

### What platform(s) does this occur on?

Android, iOS

### SDK Version (managed workflow only)

42.0.1

### Environment

```
  Expo CLI 4.9.0 environment info:
    System:
      OS: macOS 11.4
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 14.17.3 - /usr/local/bin/node
      npm: 6.14.13 - /usr/local/bin/npm
      Watchman: 4.9.0 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.10.2 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: iOS 14.5, DriverKit 20.4, macOS 11.3, tvOS 14.5, watchOS 7.4
      Android SDK:
        API Levels: 28, 29, 30
        Build Tools: 28.0.3, 29.0.2, 30.0.3
        System Images: android-30 | Google APIs Intel x86 Atom
    IDEs:
      Android Studio: 4.2 AI-202.7660.26.42.7486908
      Xcode: 12.5.1/12E507 - /usr/bin/xcodebuild
    npmPackages:
      expo: ~42.0.1 => 42.0.3
      react: 16.13.1 => 16.13.1
      react-dom: 16.13.1 => 16.13.1
      react-native: ~0.63.4 => 0.63.4
      react-native-web: ~0.13.12 => 0.13.18
      expo-updates: 0.8.1
    npmGlobalPackages:
      expo-cli: 4.9.0
    Expo Workflow: bare
```

### Reproducible demo or steps to reproduce from a blank project

Reproducible demo: https://github.com/RodolfoGS/issue-expo-updates-42
1. `git clone git@github.com:RodolfoGS/issue-expo-updates-42.git`
2. `cd issue-expo-updates-42`
3. `npm install`
4. `expo export --dump-sourcemap --public-url __YOUR_PUBLIC_URL__` (in order to update `EXPO_UPDATE_URL`)
5. Open project on Xcode
6. Edit scheme -> Build Configuration -> Release
7. Run on your device
8. `expo export --dump-sourcemap --public-url __YOUR_PUBLIC_URL__`
9. Upload `dist/` content to your server
10. Tap on `Check Update` button on the example app

**Comment by RodolfoGS** [comment]
**I solved the issue.**

My server was responding with the header `Cache-Control: max-age=31536000,public`. I used that for years, but seem that in the latest version of `expo-updates` something was changed and now the device read that header and save the `ios-index.json` and `android-index.json` files on the device's cache.

I'll close the issue, if someone have this issue, try removing or change the `Cache-Control` header from your server to avoid the device's cache.
